cmake/modules/SwiftSharedCMakeConfig.cmake

```
@@ -58,7 +58,7 @@ macro(swift_common_standalone_build_config_llvm product)
     fix_imported_targets_for_xcode("${LLVM_EXPORTED_TARGETS}")
   endif()
 
-  if(NOT CMAKE_CROSSCOMPILING AND NOT SWIFT_CROSS_COMPILING)
+  if(NOT CMAKE_CROSSCOMPILING)
```

I'm wondering if it would be best to add `AND NOT SWIFT_PREBUILT_CLANG` here as well -- this variable still seems used in a few places, and it would be consistent to the way we are setting `SWIFT_NATIVE_CLANG_TOOLS_PATH` (at least with regard to the toolchain preset)

```suggestion
  if(NOT CMAKE_CROSSCOMPILING) AND NOT SWIFT_PREBUILT_CLANG
```

---

I considered what to do here and found only [one place the native llvm path is used after this pull, as a path passed in to lit when running the tests](https://github.com/apple/swift/blob/d7dfa3e942dbab243e56af96ff08fdcc1cfd2c16/test/CMakeLists.txt#L136).

Since `SWIFT_PREBUILT_CLANG` is only keyed to `SWIFT_NATIVE_CLANG_TOOLS_PATH`, it is possible that some dev might pass in a native llvm path and no native clang path, but adding this check you suggest wouldn't pass that native llvm path to the tests then. That's an unlikely scenario, but ideally there'd be a separate `SWIFT_PREBUILT_LLVM` keyed off the native llvm path and then you'd check that here.

Since Mishal didn't bother adding a similar variable for clang when he added `SWIFT_CROSS_COMPILING`, I figured it's not a big deal and left it out, could add another LLVM variable now if you like.

---

Thanks for the explanation -- given that the toolchain has successfully built, I think this is not needed for now.